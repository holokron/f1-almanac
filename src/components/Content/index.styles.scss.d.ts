export const contentRow: string;
export const contentPageHeader: string;
