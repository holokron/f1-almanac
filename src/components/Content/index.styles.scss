.contentRow {
    padding: 1rem;
}

.contentPageHeader {
    margin-bottom: 1rem;
}