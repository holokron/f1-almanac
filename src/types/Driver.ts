export interface DriverListItem {
  driverId: string
  permanentNumber?: string
  code?: string
  url: string
  givenName: string
  familyName: string
  dateOfBirth: string
  nationality: string
}

export type DriversList = DriverListItem[]
