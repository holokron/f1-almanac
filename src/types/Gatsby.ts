export interface SiteMetadata {
  author: string
  copyright: string
  description: string
  keywords: string
  title: string
  version: string
}
