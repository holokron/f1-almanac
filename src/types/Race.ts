import { Circuit } from "./Circuit"

export interface Race {
  season: string
  round: string
  url: string
  raceName: string
  Circuit: Circuit
  date: string
  time: string
}

export type RacesList = Race[]
