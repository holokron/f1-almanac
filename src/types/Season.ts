export interface SeasonListItem {
  season: string
  url: string
}

export type SeasonsList = SeasonListItem[]
