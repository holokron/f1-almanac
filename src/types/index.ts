export * from "./Breadcrumb"
export * from "./Circuit"
export * from "./Driver"
export * from "./Location"
export * from "./Race"
export * from "./Season"
export * from "./SiteData"
export * from "./Standings"
export * from "./Team"
