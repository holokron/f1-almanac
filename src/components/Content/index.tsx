import React from "react"
import { Row, PageHeader } from "antd"
import * as styles from "./index.styles.scss"
import { Breadcrumb } from "../../types"
import { Link, navigate } from "gatsby"

interface ContentProps {
  breadcrumbs: Breadcrumb[]
  children: React.ReactNode
  title: string
  navigateBack?: string
}

export default React.memo(
  ({
    breadcrumbs,
    children,
    title,
    navigateBack,
  }: ContentProps): React.ReactElement<ContentProps> => (
    <Row className={styles.contentRow}>
      <PageHeader
        onBack={navigateBack ? () => navigate(navigateBack) : undefined}
        className={styles.contentPageHeader}
        title={title}
        breadcrumb={{
          routes: breadcrumbs.map((item: Breadcrumb) => ({
            breadcrumbName: item.name,
            path: item.path,
          })),
          itemRender: (
            route: any,
            params: any,
            routes: Array<any>,
            paths: Array<string>,
          ) => <Link to={route.path}>{route.breadcrumbName}</Link>,
        }}
      />
      {children}
    </Row>
  ),
)
