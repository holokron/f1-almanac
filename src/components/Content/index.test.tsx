import React, { ReactNode } from "react"
import renderer from "react-test-renderer"
import Content from "."
import { Breadcrumb } from "../../types"

const breadcrumbs: Breadcrumb[] = [
  {
    name: "Foo",
    path: "/foo",
  },
  {
    name: "Bar",
    path: "/bar",
  },
]

describe("components/Content", (): void => {
  it.each([["Test", "Foo Bar", "/foo"]])(
    "data: %p",
    (children: ReactNode, title: string, navigateBack: string = ""): void => {
      const tree = renderer.create(
        <Content
          breadcrumbs={breadcrumbs}
          title={title}
          navigateBack={navigateBack}
        >
          {children}
        </Content>,
      )

      expect(tree.toJSON()).toMatchSnapshot()
    },
  )
})
