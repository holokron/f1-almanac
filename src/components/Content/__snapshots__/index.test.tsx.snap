// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Content data: "Test" 1`] = `
<div
  className="ant-row contentRow"
>
  <div
    className="ant-page-header contentPageHeader"
  >
    <div
      className="ant-breadcrumb"
    >
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="/foo"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Foo
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="/bar"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Bar
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </div>
    <div
      className="ant-page-header-title-view"
    >
      <span
        className="ant-page-header-title-view-title"
      >
        Foo Bar
      </span>
    </div>
  </div>
  Test
</div>
`;
